src/ProxyClient/HttpProxyClient.php

```diff
@@ -50,17 +50,17 @@ abstract class HttpProxyClient implements ProxyClient
      *
      * @param Dispatcher          $httpDispatcher Helper to send HTTP requests to caching proxy
      * @param array               $options        Options for this client
-     * @param RequestFactory|null $messageFactory Factory for PSR-7 messages. If none supplied,
+     * @param RequestFactoryInterface|null $messageFactory Factory for PSR-7 messages. If none supplied,
      *                                            a default one is created
      */
     public function __construct(
         Dispatcher $httpDispatcher,
         array $options = [],
-        RequestFactory $messageFactory = null
+        RequestFactoryInterface $messageFactory = null
     ) {
         $this->httpDispatcher = $httpDispatcher;
         $this->options = $this->configureOptions()->resolve($options);
-        $this->requestFactory = $messageFactory ?: MessageFactoryDiscovery::find();
+        $this->requestFactory = $messageFactory ?: Psr17FactoryDiscovery::findRequestFactory();
```

Comment:
we should fallback to MessageFactoryDiscovery::find if the Psr17FactoryDiscovery does not find anything.